src/shared/PreApprovalRequest/DetailsHelper.js

```diff
@@ -1,8 +1,9 @@
 import { DefaultDetails } from './DefaultDetails';
 import { Code105Details } from './Code105Details';
 
-export function getDetailComponent(code, robustAccessorial) {
+export function getDetailComponent(code, robustAccessorial, initialValues) {
   code = code ? code.toLowerCase() : '';
-  if (code && code.startsWith('105') && !code.startsWith('105d') && robustAccessorial) return Code105Details;
+  if (initialValues && !initialValues.crate_dimensions) return DefaultDetails;
+  if ((code && code.startsWith('105b')) || (code.startsWith('105e') && robustAccessorial)) return Code105Details;
```

Comment:
Couple things:

- When checking the code value, we should be checking it together. Not separately.
- We can drop the truthy `code` from the beginning as well since we're checking for the 105b/e string value and we're setting the value from the beginning.
- We also want to be able to use the `robustAccessorial` flag so when check for dimensions in the 105 logic as well
- Add some tests around this functionality

Something like this:
```suggestion
const hasDimensions = get(initialValues, 'crate_dimensions') ? true : false; //using lodash here
  if ((code.startsWith('105b') || code.startsWith('105e')) && hasDimensions && robustAccessorial) return Code105Details;
```